package/MDAnalysis/analysis/secondary_structure/dssp.py

```diff
@@ -0,0 +1,254 @@
+    ss_simple: :class:`numpy.ndarray` of shape (n_frames, n_residues)
+        Simplified secondary structure names
+    ss_counts: dict of {code: :class:`numpy.ndarray` of shape (n_frames,)}
+        Dictionary that counts number of residues with each secondary
+        structure code, for each frame
+    simple_counts: dict of {name: :class:`numpy.ndarray` of shape (n_frames,)}
+        Dictionary that counts number of residues with each simplified 
+        secondary structure, for each frame
+    ss_mode: :class:`numpy.ndarray` of shape (n_residues,)
+        The most common secondary structure for each residue
+    simple_mode: :class:`numpy.ndarray` of shape (n_residues,)
+        The most common simple secondary structure for each residue
+    ss_codes_to_names: dict of {code: name}
+        Dictionary converting each single-letter code to the full name of
+        the secondary structure
+    ss_codes_to_simple: dict of {code: name}
+        Dictionary converting each single-letter code to simplified 
+        secondary structures
+    atomgroup: :class:`~MDAnalysis.core.groups.AtomGroup`
+        An ordered AtomGroup of backbone atoms
+    nco_indices: :class:`numpy.ndarray` of shape (n_residues, 3)
+        An array of the atom indices of N, C, O atoms, *relative* to the ``atomgroup``.
+        An index is -1 if it is not found in the residue.
+    ca_indices: :class:`numpy.ndarray` of shape (n_residues,)
+        An array of the atom indices of CA atoms, *relative* to the ``atomgroup``.
+        An index is -1 if it is not found in the residue.
+    is_proline: :class:`numpy.ndarray` of shape (n_residues,)
+        An integer array indicating whether a residue is a proline. 1 is True,
+        0 is False.
+    not_protein: :class:`numpy.ndarray` of shape (n_residues,)
+        A boolean array indicating whether a residue has all of the N, C, O, CA atoms.
+    chain_ids: :class:`numpy.ndarray` of shape (n_residues,)
+        An integer array representing the index of the chainID or segment each residue 
+        belongs to.
+
+
+    Example
+    -------
+
+    ::
+
+        import MDAnalysis as mda
+        from MDAnalysis.tests.datafiles import PSF, DCD
+        from MDAnalysis.analysis import secondary_structure as ss
+
+        u = mda.Universe(PSF, DCD)
+        dssp = ss.DSSP(u, select='backbone', o_name='O O1',
+                       add_topology_attr=True)
+        dssp.run()
+        ss0 = u.residues[0].secondary_structure
+        print('The first residue has secondary structure {}'.format(ss0))
+
+    The full secondary structure codes can be accessed at :attr`DSSP.ss_codes`.
+
+
+    """
+
+    def __init__(self, universe, select='backbone', verbose=False,
+                 add_topology_attr=False, c_name='C', o_name='O O1',
+                 n_name='N', ca_name='CA', pro_name='PRO'):
+        # TODO: implement this on its own w/o mdtraj?
+        try:
+            from mdtraj.geometry._geometry import _dssp
```

Comment:
Would it be worthwhile to copy the code and include it here (under LGPL) and of course leave all the citations?

Reply:
I suppose so, if this is better than importing MDTraj?